src/index.ts

```diff
@@ -2288,12 +2288,12 @@ export interface GetOrCreateParams<T extends Resource = AnyResource>
 	extends Omit<Params<T>, 'method' | 'url'> {
 	id: ResourceAlternateKey<T['Read']>;
 	resource: string;
-	body: T['Write'];
+	body: Partial<T['Write']>;
```

Comment:
Slightly confused with this one @Page- ? for the upsert it makese sense to be partial, but for a Post shouldn't it match T['Write'] exactly?

Reply:
No because fields can have defaults, ideally we would know all fields that have defaults (via schema or hooks) and make only those optional but for now this is imo the best trade off to get usable, useful typings that help do the right thing without blocking it. It's definitely a place that can be improved though